app/models/user.rb

```diff
@@ -0,0 +1,33 @@
+require 'bcrypt'
+class User
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :username, String
+  property :email, String, required: true, unique: true, message: 'This email is already taken'
+
+  property :password_digest, Text
+
+  has n, :peeps, through: Resource
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(email, password)
+    user = User.first(email: email)
+
+    if user && BCrypt::Password.new(user.password_digest) == password
+      user
+    else
+      nil
+    end
+  end
+
+  validates_confirmation_of :password
```

Comment:
Prefer the new style validations `validates :column, confirmation: value` over `validates_confirmation_of`.